main.go

```diff
@@ -173,19 +176,36 @@ func ndkVersion(ndkPath string) string {
 	return ""
 }
 
-func currentNDKHome() string {
-	if v := os.Getenv(androidNDKHome); v != "" {
+// https://github.com/android/ndk-samples/wiki/Configure-NDK-Path
+// https://developer.android.com/tools/variables
+func currentNDKHome(envRepo env.Repository, sys fs.FS, requestedNDKVersion string) string {
+	if v := envRepo.Get(androidNDKHome); v != "" {
+		// $ANDROID_NDK_HOME is old and AGP no longer takes it into account,
+		// but it's an explicit path, so use it if it's set on the system.
 		return v
 	}
-	if v := os.Getenv("ANDROID_HOME"); v != "" {
-		// $ANDROID_HOME is deprecated
-		return filepath.Join(v, "ndk-bundle")
+	if androidHome := envRepo.Get("ANDROID_HOME"); androidHome != "" {
+		// The most modern way is to install NDK versions side-by-side at $ANDROID_HOME/ndk/version
+		// This is what `sdkmanager` does when installing a specific version (`sdkmanager "ndk;26.3.11579264"`).
+		ndkPath := filepath.Join(androidHome, "ndk", requestedNDKVersion)
+		_, err := fs.Stat(sys, ndkPath)
+		if err == nil {
+			return ndkPath
+		}
+
+		// $ANDROID_HOME/ndk-bundle is deprecated, it's the non-side-by-side install location
+		// installed by `sdkmanager ndk-bundle`
+		ndkBundlePath := filepath.Join(androidHome, "ndk-bundle")
+		_, err = fs.Stat(sys, ndkBundlePath)
+		if err != nil {
+			return ndkBundlePath
+		}
 	}
-	if v := os.Getenv("ANDROID_SDK_ROOT"); v != "" {
-		// $ANDROID_SDK_ROOT is preferred over $ANDROID_HOME
+	if v := envRepo.Get("ANDROID_SDK_ROOT"); v != "" {
+		// $ANDROID_SDK_ROOT is deprecated, so it's lower in priority than $ANDROID_HOME
```

Comment:
Shouldn't the logic from lines 190-194 be also here?
ANDROID_SDK_ROOT is an alternative to ANDROID_HOME

Reply:
That's what I thought too, but the latest doc say something else:
https://developer.android.com/tools/variables
<img width="841" alt="image" src="https://github.com/bitrise-steplib/steps-install-missing-android-tools/assets/1694986/dd84c7ed-99d7-409a-ba0d-56dd9297be31">

Reply:
But yes, even if it's deprecated, it's probably a good idea to handle the two env vars the same way. Thank you!